packages/snap/tests/rpc/onRpcRequest.spec.ts

```diff
@@ -1624,4 +1618,180 @@ describe('onRpcRequest', () => {
+          id: 'test-id',
+          jsonrpc: '2.0',
+          method: 'createVC',
+          params: {
+            minimalUnsignedCredential: unsignedMinimalCredential,
+            proofFormat: 'EthereumEip712Signature2021',
+          },
+        },
+      })) as Result<boolean>;
+
+      if (isError(vc)) {
+        throw vc.error;
+      }
+
+      console.log(vc.data);
+      expect(vc.data).toContainKey('proof');
+      expect((vc.data as unknown as VerifiableCredential).proof.type).toBe(
+        'EthereumEip712Signature2021'
+      );
+      expect.assertions(2);
+    });
+    // TODO fix JSON-LD VC creation
+    // it('should succeed creating a JSON-LD VC', async () => {
+    //   const vc = (await onRpcRequest({
+    //     origin: 'localhost',
+    //     request: {
+    //       id: 'test-id',
+    //       jsonrpc: '2.0',
+    //       method: 'createVC',
+    //       params: {
+    //         minimalUnsignedCredential: unsignedMinimalCredential,
+    //         proofFormat: 'lds',
+    //       },
+    //     },
+    //   })) as Result<boolean>;
+
+    //   if (isError(vc)) {
+    //     throw vc.error;
+    //   }
+
+    //   console.log(vc.data);
+    //   expect(vc.data).toContainKey('proof');
+    //   expect((vc.data as unknown as VerifiableCredential).proof.type).toBe(
+    //     'EthereumEip712Signature2021'
+    //   );
+    //   expect.assertions(2);
+    // });
+
+    // TODO: throw error for invalid VCs
+    // it('should fail creating a EIP VC', async () => {
+    //   const fakeVC = {
+    //     data: 'jhello',
+    //   };
+
+    //   const vc = (await onRpcRequest({
+    //     origin: 'localhost',
+    //     request: {
+    //       id: 'test-id',
+    //       jsonrpc: '2.0',
+    //       method: 'createVC',
+    //       params: {
+    //         minimalUnsignedCredential: fakeVC,
+    //         proofFormat: 'EthereumEip712Signature2021',
+    //       },
+    //     },
+    //   })) as Result<boolean>;
+
+    //   if (isError(vc)) {
+    //     throw vc.error;
+    //   }
+
+    //   console.log(vc.data);
+    //   expect(vc.data).toContainKey('proof');
+    //   expect((vc.data as unknown as VerifiableCredential).proof.type).toBe(
+    //     'EthereumEip712Signature2021'
+    //   );
+    //   expect.assertions(2);
+    // });
+    it('should succeed creating and saving a JWT VC', async () => {
+      const vc = (await onRpcRequest({
+        origin: 'localhost',
+        request: {
+          id: 'test-id',
+          jsonrpc: '2.0',
+          method: 'createVC',
+          params: {
+            minimalUnsignedCredential: unsignedMinimalCredential,
+            proofFormat: 'jwt',
+            options: {
+              save: true,
+            },
+          },
+        },
+      })) as Result<boolean>;
+
+      if (isError(vc)) {
+        throw vc.error;
+      }
+
+      expect(vc.data).toContainKey('proof');
+      expect((vc.data as unknown as VerifiableCredential).proof.type).toBe(
```

Comment:
You can add the types in the `Result<>` part and then you don't need to case them in the expect block